lhotse/dataset/transforms/cat.py

```diff
@@ -0,0 +1,79 @@
+from typing import Optional, Sequence
+
+from lhotse import CutSet
+from lhotse.cut import AnyCut
+from lhotse.utils import Seconds
+
+
+class CutCat:
+    """
+    A transform on batch of cuts (``CutSet``) that concatenates the cuts to minimize the total amount of padding;
+    e.g. instead of creating a batch with 40 examples, we will merge some of the examples together
+    adding some silence between them to avoid a large number of padding frames that waste the computation.
```

Comment:
Bear in mind that this could be a little dangerous to use especially if there is batchnorm, as it messes with statistical independence (especially if the data comes from multiple sources with different length distributions).
You might want to warn about that.

Reply:
Good point, we might want to try removing that from snowfall expts. I am planning to implement a "BucketingCutSampler" soon that will group cuts based on histograms of durations and draw from a single bucket for any individual batch, to avoid wasting frames on padding as an alternative to this approach (should work great with dynamic batch size). We can check in snowfall then.

Reply:
I assume there is an option to have truly random sampling?  Would be nice if someone could try this in snowfall, it might slightly impove the results due to interaction with batchnorm (and also optimization issues if the batches' order is no randomized).

Reply:
I will try to test it in snowfall.

Reply:
> I assume there is an option to have truly random sampling?

As a result of this PR, the shuffling is "truly random" now (before it was only "truly random inside partitions")